### PR TITLE
Add new colors for WaveColor

### PR DESCRIPTION
**Pull request by ClausMartini**
This Audacity fork needs a facelift in order to stand out (in my opinion). And what better way to do so than improving the current wave color system? Four colors is probably good enough for the basic to mid level user. For advanced users with large numbers of tracks, this isn't good enough.
So this commit adds 20 colors to Wave Color selection.
There are 2 file patches: AllThemeResources.h and TrackArtist.cpp.

- [x] I made sure the code compiles on my machine.
- [x] I made sure there are no unnecessary changes in the code.
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving.
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?".
- [x] I hereby reaffirm that I am licensing my contribution under the GNU General Public License v2.0 or later (`GPL-2.0-or-later`).

### CODE REVIEW

File: src/AllThemeResources.h [hunk 1]
```diff
@@ -229,11 +228,16 @@ from there.  Audacity will look for a file called "Pause.png".
    DEFINE_IMAGE( bmpBottomFrequencyCursor, wxImage( 32, 32 ), wxT("BottomFrequencyCursor"));
    DEFINE_IMAGE( bmpTopFrequencyCursor, wxImage( 32, 32 ), wxT("TopFrequencyCursor"));
    DEFINE_IMAGE( bmpBandWidthCursor, wxImage(32, 32), wxT("BandWidthCursor"));
-   DEFINE_IMAGE( bmpSubViewsCursor, wxImage(32, 32), wxT("SubViewsCursor"));
```
Comment (AnErrupTion) [inline src/AllThemeResources.h:232]: I'm curious, why did you remove this line? It looked perfectly fine.

File: src/AllThemeResources.h [hunk 2]
```diff
@@ -229,11 +228,16 @@ from there.  Audacity will look for a file called "Pause.png".
...
 
    //SET_THEME_FLAGS(  resFlagNewLine  );
 
+// DA: The logo with name xpm has a different width.
+#ifdef EXPERIMENTAL_DA
```
Comment (AnErrupTion) [inline src/AllThemeResources.h:235]: Why did you add EXPERIMENTAL_DA stuff? We're trying to remove anything related to DA so this doesn't help.

**Review by AnErrupTion — CHANGES_REQUESTED** [review]

File: src/TrackArtist.cpp [hunk 3]
```diff
@@ -400,7 +489,7 @@ void TrackArt::DrawBackgroundWithSelection(
    dc->SetPen(*wxTRANSPARENT_PEN);
    if (track->GetSelected() || track->IsSyncLockSelected())
    {
-      // Rectangles before, within, after the selection
+      // Rectangles before, within, after the selction
```
Comment (matt-hu) [inline src/TrackArtist.cpp:492]: Again, another typo being introduced

File: src/TrackArtist.cpp [hunk 4]
```diff
@@ -266,7 +355,7 @@ void TrackArtist::UpdatePrefs()
 // two steps down for every one across. This creates a pattern that repeats in
 // 5-step by 5-step boxes. Because we're only drawing in 5/25 possible positions
 // we have a grid spacing somewhat smaller than the image dimensions. Thus we
-// achieve lower density than with a square grid and eliminate edge cases where
+// acheive lower density than with a square grid and eliminate edge cases where
```
Comment (matt-hu) [inline src/TrackArtist.cpp:358]: Being a little picky here, but this introduces a typo in the comment

**Review by matt-hu — CHANGES_REQUESTED** [review]

**Comment by ClausMartini** [comment]
@AnErrupTion A mistake. As you can see, this is work in progress. Should have said that in the description. Sorry.
Just thought it'd make a fine addition to Audacium.
So feel free to make all necessary changes as you see fit.
@matt-hu Why would you request typos?

File: src/AllThemeResources.h (hunk 2 above)
Comment (ClausMartini) [reply src/AllThemeResources.h:235]: Sorry. Feel free to come up with something better.

**Comment by matt-hu** [comment]
@ClausMartini I'm not requesting them, you have made them, I'm pretty sure

**Comment by ClausMartini** [comment]
@matt-hu You're right. Sorry. My dyslexia. 😀

**Comment by matt-hu** [comment]
@ClausMartini no worries! Tbh, I'm probably slightly dyslexic and had to look them up myself to see if that was the right spelling.

Oh and yes, you are very much right in that it needs a facelift

**Comment by AnErrupTion** [comment]
@ClausMartini Could you just answer the requested changes I put? There's a line you removed I'm curious about, that's all. And if you could remove reference to EXPERIMENTAL_DA it would be nice.

File: src/AllThemeResources.h (hunk 2 above)
Comment (AnErrupTion) [reply src/AllThemeResources.h]: I don't want to be rude but you can simply update the file and removing the stuff, y'know? ;)

Comment (AnErrupTion) [reply src/AllThemeResources.h]: Ah, my bad. You should keep only the line `#define LOGOWITHNAME_WIDTH 506`.

**Comment by ClausMartini** [comment]
@AnErrupTion Done!

**Comment by AnErrupTion** [comment]
My bad, you only need to keep the line ``#define LOGOWITHNAME_WIDTH 506``. For the other line, I was only curious about it. You didn't have to add it again, I'm sure you had a reason you deleted it! If you could just fix that and explain me I'll merge :)

**Comment by ClausMartini** [comment]
Nah, it should be fine now.
I'm sure I removed it by mistake in the first place.

**Comment by AnErrupTion** [comment]
Just the last thing, the very last before I can merge : You've removed ``#ifdef EXPERIMENTAL_DA``, however you didn't remove the last 3 lines so it will cause errors. Could you simply only keep the line ``#define LOGOWITHNAME_WIDTH 506`` and not the 2 other lines? Thanks if you do :)

**Comment by AnErrupTion** [comment]
Oh damn, forgot to say to remove the ``#endif`` too... this isn't your fault, don't worry! This is actually the last thing.

**Comment by ClausMartini** [comment]
Right. Sorry. I've been sitting on the computer all day, so forgive me if I seem completely spaced out. 😀
Thanks both of you @matt-hu and @AnErrupTion .

**Comment by AnErrupTion** [comment]
Thanks for your contribution!

**Comment by ClausMartini** [comment]
You're very welcome.
